README.markdown

```diff
@@ -449,6 +449,22 @@ If you want to stop processing jobs, but want to leave the worker running
 (for example, to temporarily alleviate load), use `USR2` to stop processing,
 then `CONT` to start it again.
 
+#### Signals on Heroku
+
+When shutting down processes, Heroku sends every process a TERM signal at the
+same time. By default this causes an immediate shutdown of any running job
+leading to frequent `Resque::TermException` errors.  For short running jobs, a simple
+solution is to give a small amount of time for the job to finish
+before killing it.
+
+To accomplish this set the following environment variables:
+
+* `RESQUE_PRE_SHUTDOWN_TIMEOUT` - The time between the parent receiving a shutdown signal (TERM by default) and it sending that signal on to the child process. Designed to give the child process
+time to complete before being forced to die.
+
+* `RESQUE_SHUTDOWN_SIGNAL` - The signal sent to the child process
```

Comment:
We do not seem to need it, this is the setup we have now. 

`resque:        env RESQUE_PRE_SHUTDOWN_TIMEOUT=25 RESQUE_TERM_TIMEOUT=2 TERM_CHILD=1 INTERVAL=1 QUEUE=high,medium,mail,low bundle exec rake resque:work`

It allows us 25 seconds for jobs to finish, then 2 seconds (as far as we understand the code) for the failure handling to finish. We use that time to requeue all jobs that haven't finished by catching TermException in `Job#fail`(so that we don't have to catch it in all of our jobs).

We haven't actually run it on heroku yet though, but our old environment is setup to send the same signals. TERM first, then wait 30 seconds and KILL if it's still running. We're in the process of migrating to heroku.